### ISSUE TITLE
Update composer compatibility deprecation

### ISSUE THREAD

**Issue by Wouter0100** [Elgg/Elgg#9855]
Elgg seems to require something that's deprecated in Composer:

```
Deprecation Notice: The callback \Elgg\Composer\PostUpdate::execute declared at [removed]/vendor/elgg/elgg/engine/classes/Elgg/Composer/PostUpdate.php accepts a Composer\Script\Event but post-package-update events use a Composer\Script\PackageEvent instance. Please adjust your type hint accordingly, see https://getcomposer.org/doc/articles/scripts.md#event-classes in phar:///usr/local/bin/composer/src/Composer/EventDispatcher/EventDispatcher.php:303
```

This notice is outputted every time a package is updated (`\Elgg\Composer\PostUpdate::execute`).

**Comment by juho-jaakkola** [comment]
This line needs to be updated: https://github.com/Elgg/Elgg/blob/2.0/engine/classes/Elgg/Composer/PostUpdate.php#L17

So that the execute function takes instance of  `PackageEvent` instead of `Event`.

**Comment by juho-jaakkola** [comment]
Same here: https://github.com/Elgg/Elgg/blob/2.0/engine/classes/Elgg/Composer/PostInstall.php#L19

**Comment by jdalsem** [comment]
Fixed in #11027
